### ISSUE TITLE
[bug] [SDK] project_id not found

### ISSUE THREAD

**Issue by samuelstroschein** [opral/monorepo#2070]
### Problem

From https://discord.com/channels/897438559458430986/1198317791799935107

<img width="750" alt="CleanShot 2024-01-21 at 00 36 44@2x" src="https://github.com/opral/monorepo/assets/35429197/d35fcfe3-7384-4ef7-938c-2d0b4d1b06b7">

### Expected behavior

If no project id can be generated, `loadProject` should not throw (until https://github.com/opral/monorepo/issues/2069 is implemented)

### Reproduction

not provided

**Comment by samuelstroschein** [comment]
cc @janfjohannes who implemented the logic. he might know what's causing this bug

**Comment by felixhaeberle** [comment]
@janfjohannes thanks for the ping, will update the credentials and set it in 1PW.

**Comment by felixhaeberle** [comment]
@janfjohannes Updated ✅ Please be aware that my git username/mail are being set in the windows instance in order to have git function properly (don't know – I haven't pushed there but it cried at me).

**Comment by jldec** [comment]
@janfjohannes I'll try to repro this since I've been working in the loadProject code for #2066 
Please let me know if you have a theory about where this might be happening.

I did find this mention of pursiny-sveltekit in https://github.com/KraXen72/tech-support-wiki/blob/master/docs/nodejs/pnpm-link.md - I wonder if perhaps `pnpm link` created a symlink which we're not following (just a theory 🤔) 
cc: @KraXen72

**Comment by KraXen72** [comment]
@jldec pursiny-sveltekit is a site i built for a client, which is not open-source. while i do not have any limitations on sharing snippets or files from that repo imposed by the client, i don't think they'd be happy if i shared the entire source code.  
I have some `pnpm link`s in that repo, but it's nothing to do with inlang (i just pnpm linked a sveltekit adapter)

**Comment by samuelstroschein** [comment]
The bug might (likely) be caused by hardcoded POSIX paths (once again). I opened https://github.com/opral/monorepo/issues/2075

**Comment by felixhaeberle** [comment]
@jldec is there a case when the project id can be `undefined` but the project is returned? I'm currently working with the `source-code/end-to-end-tests/inlang-nextjs` project in the vs code extension and there, I get an `undefined` project id. Just to confirm this is nothing I have to worry about.

**Comment by samuelstroschein** [comment]
@felixhaeberle yes there is a case that `undefined` is returned until https://github.com/opral/monorepo/pull/2066#pullrequestreview-1836293942 is merged

**Comment by jldec** [comment]
FYI for reference
https://github.com/opral/monorepo/pull/2066#issuecomment-1904335783

**Comment by jldec** [comment]
@KraXen72 - I'm afraid we have not been able to reproduce the issue you reported.
Is there any way you could share a repo on githb with the setup you are using?
(Please strip out any customer-specific files) 
I you could do this, we could try again to reproduce the error you are seeing on Windows 
Thanks
jldec @ inlang

**Comment by jldec** [comment]
See details of attempted repro in #2075

**Comment by samuelstroschein** [comment]
@jldec if we can't repro. let's close this as "can't repro" and move on [until this might re-emerge]

**Comment by KraXen72** [comment]
I'll try updating all inlang packages to latest version, and then if it still doesen't work i'll send the repo in one of you guys' dms on discord - which one should i send it to? thanks.

**Comment by samuelstroschein** [comment]
> in one of you guys' dms on discord - which one should i send it to? thanks.

send it to @jldec

**Comment by KraXen72** [comment]
changing the folder to explicitly say 'project.inlang' instead of 'pursiny-sveltekit.inlang' fixed the issue. For now this is an acceptable workaround, but if paraglide wants to support arbitrarily named .inlang directories, this might be something to look into. 

Honestly sorry I didn't try this sooner, even if it did work with the previous setup in the past.

**Comment by samuelstroschein** [comment]
> changing the folder to explicitly say 'project.inlang' instead of 'pursiny-sveltekit.inlang' fixed the issue.

THis is a bug. 

@jldec can you investigate?

**Comment by samuelstroschein** [comment]
maybe related https://github.com/opral/monorepo/issues/2101

**Comment by jldec** [comment]
@KraXen72 - I'm trying to reproduce your error and have a couple of questions

1. is your project git initiailzed (does it have a .git directory?)
2. when do you see the error that you reported? Are you triggering a compile or using the IDE extension?

**Comment by KraXen72** [comment]
yes i have a .git, i use the cli's compile

**Comment by KraXen72** [comment]
i'll send my package json

**Comment by jldec** [comment]
thanks @KraXen72 - could you also share your vite-config (I assume you had a matching project directory `pursiny-sveltekit.inlang` in there)

**Comment by KraXen72** [comment]
[package.json](https://github.com/opral/monorepo/files/14038149/package.json)
[settings.json](https://github.com/opral/monorepo/files/14038195/settings.json) for inlang
vite.config.ts - this is after i chaged everything from 'pursiny-sveltekit.inlang' to 'project.inlang' to get it to work. to get the setup i had previously, do a search&replace on the vite config & package.json, also rename the directory to `pursiny-sveltekit.inlang`.
```ts
import { defineConfig, normalizePath } from 'vite';
import { resolve } from 'node:path';

import copy from 'rollup-plugin-copy'
import { sveltekit } from '@sveltejs/kit/vite';
import ViteRestart from 'vite-plugin-restart';
import { paraglide } from '@inlang/paraglide-js-adapter-vite';
import UnoCSS from 'unocss/vite';


export default defineConfig(({ command }) => {
	const config = {
		plugins: [
			UnoCSS(),
			sveltekit(),
			paraglide({
				project: './project.inlang',
				outdir: './src/paraglide',
			}),
			ViteRestart({ restart: 'mdsvex.config.js' }),
		],
		server: {
			fs: {
				allow: ['posts'] // required by mdsvex & our strategy
			},
		},
		optimizeDeps: {
			exclude: ['@inlang/paraglide-js']
		},
		ssr: {
			noExternal: ['@inlang/paraglide-js']
		},
	}
	if (command === 'build') {
		const copyPlugin = copy({
			targets: [
				{ 
					src: ['db/*.htaccess', 'db/schema.json', 'db/*SCINDEX.json', 'db/vtmeta.yml'], 
					dest: normalizePath(resolve('.svelte-kit', 'adapter-iis', 'db'))
				},
				{
					src: ['static/krpano_svelte_actions.xml'],
					dest: normalizePath(resolve('static-prod'))
				}
			],
			hook: 'writeBundle',
			overwrite: true
		})

		config.plugins.push(copyPlugin)
	}

	return config

});
```
this is how my directories are set up:
![image](https://github.com/opral/monorepo/assets/21956756/de57fb20-0392-4e36-9ebb-2f5b3adc8afc)

**Comment by jldec** [comment]
cc: @LorisSigrist - copying you here just in case something obvious jumps out:
tl;dr: `paraglide-js compile` is erroring, looking in "project.inlang" instead of the actual "pursiny-sveltekit.inlang" directory.

Could the vite adapter be doing something differently vs. sveltekit adapter or
Could the versions mentioned in the OP explain why this error happens?

**Comment by jldec** [comment]
I just managed to repro something similar 🎉 
(not 100% sure if it's the same bug)

The project is in the `vite-adapter` branch of [jldec/paraglide](https://github.com/jldec/paraglide/tree/vite-adapter)
It has latest paraglide-js and paraglide-js-adapter-vite and no project_id file in the `my-project.inlang` directory.

```sh
git clone https://github.com/jldec/paraglide.git jldec-paraglide
cd jldec-paraglide
git checkout vite-adapter
pnpm install
```
```
$ pnpm install
Packages: -2
--
Progress: resolved 262, reused 228, downloaded 0, added 0, done

> paraglide@0.0.1 postinstall /Users/jldec/inlang/paraglide
> paraglide-js compile --project ./my-project.inlang

ℹ [paraglide] Compiling inlang project at "./my-project.inlang".                                                                                                         
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: ENOENT: no such file or directory, open '/Users/jldec/inlang/paraglide/my-project.inlang/project_id'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/jldec/inlang/paraglide/my-project.inlang/project_id'
}

Node.js v20.11.0
```

**Comment by jldec** [comment]
Thanks for confirming that @KraXen72

**Comment by samuelstroschein** [comment]
@opral/team please do not assume or default to `project.inlang`, see https://github.com/opral/monorepo/issues/2070#issuecomment-1908823467

- we defaulted to `project.inlang` for a migration 
- do not assume `project.inlang` anywhere else. look for available projects, let the user select the project. 

---

@jldec that's why I am hesitant to drop the `--project` flag from the CLIs for example.

**Comment by janfjohannes** [comment]
@samuelstroschein  this is still really confusing to me, if there is a project.inlnag forlder i expect a cli to use that without me having to retype it.

**Comment by jldec** [comment]
#### status update:

I'm sorry, but I was wrong about the repro above. I still cannot reproduce this issue.

I mistakenly did not rename the folder to `my-project.inlang` when I [pushed the test branch](https://github.com/jldec/paraglide/commit/ff5d274545bdd59976dc4f63bc97d3edb9146f5c) last night. 

I have tried to recreate a project with the same package.json deps as @KraXen72 kindly shared, and the same plugins in the vite-config, and a non-default `pursiny-sveltekit.inlang` directory without a project_id. 

The project compiles properly with paraglide-js on both mac and windows.

The code is in the `pursiny-sveltekit.inlang` branch of [jldec/paraglide](https://github.com/jldec/paraglide/tree/pursiny-sveltekit.inlang) on github.

I will stop here for now, but if @KraXen72 or someone else still has this issue and can provide test repo in GitHub which reproduces the problem, I'l be happy to take another look.

**Comment by janfjohannes** [comment]
@jldec  worth checking if the user has outdated deps due to some lock file with lax version pinning in the package.

**Comment by jldec** [comment]
With @KraXen72 's help, I think we have confirmed that a `project.inlang` project path in the vite.config was very likely the root cause of the errors reported. We believe it only manifested after updating npm packages because the old vite adapter was not actively running (and messages were still being compiled manually by invoking the cli) before the update.

**Comment by jldec** [comment]
@janfjohannes - thanks for the tip about outdated deps - i used `pnpm i --frozen-lockfile` with the lockfile from the user's environment to try to force consistency

**Comment by samuelstroschein** [comment]
apparently, the bug is still not resolved https://github.com/opral/monorepo/issues/2162#issuecomment-1924162030

**Comment by jldec** [comment]
I'm confident that we figured out the mistake that led the report in this issue (error updating path in vite.config)

@2162 was unrelated to project IDs. I added https://github.com/opral/monorepo/issues/2162#issuecomment-1929321135

Regardig project IDs, please see https://github.com/opral/monorepo/issues/2101#issuecomment-1911907633
waiting for definitive way to ensure that projects have a project id. - #2069
